### ISSUE TITLE
Group feature

### ISSUE THREAD

**Issue by l2ig** [l2ig/aqua#50]
I think a group feature would be cool.
It should maybe have some of the following features.

* Prefix
* Middleware

Maybe something like:
```typescript
app.group((context) => {
    context.get(...);
    ....
});
```

**Comment by LoganTann** [comment]
I really don't think if this is the style of code intended for this project, but I love decorators like they are used in Spring

Why not trying to create this as a side project :

```ts
@Controller("hello")
export default class students {
  @get("/")
  public static getStudent(req) {
     return "hello world";
  }
  @get("/:name")
  public static getStudent(req) {
     return `hello, ${req.parameters.name} !`;
  }
}
```

**Comment by felixfong227** [comment]
This looks very NestJS-ish, maybe it might go against the idea of being a minimal framework?

**Comment by LoganTann** [comment]
> This looks very NestJS-ish, maybe it might go against the idea of being a minimal framework?

Exactly, that's why I talked about the idea of create such structure as a personal project instead of directly implementing this in aqua 😊
